api/testing_api.go

```diff
@@ -0,0 +1,57 @@
+package api
+
+import (
+	"testing"
+
+	"github.com/hashicorp/consul/sdk/testutil"
+)
+
+type ConfigCallback func(c *Config)
```

Comment:
It might not be worth publicly exporting this named symbol. `ConfigCallback` and `func(c *Config)` only differ in length by 1 character. It might be sufficient to inline it and omit the named type.

Reply:
It doesn't save much typing but I think its easier to grok having the typedef like:

```
type MyStruct struct {
  ...
  Fn ConfigCallback
  ...
}

func doSomething(fn ConfigCallback) error {
...
}
```

Instead of

```
type MyStruct struct {
   ...
   Fn func(c *Config)
   ...
}

func doSomething(fn func(c *Config)) {

}
```

Especially in the case of a function definition its easier to read without all the extra parens necessary.

Reply:
The symbol appears to be exclusively for tests, but in order to be used by `api/watch` needs to be public which isn't great from the perspective of someone skimming the godoc output for `api`.

Reply:
I see the argument there. To minimize what needs to be exported we could just inline the def to keep the godoc cleaner.